atlp-rwanda/cabal-phantom-backend
# #P6PsRgMK CRUD Operations for Buses

### **What about this PR?**
This PR has CRUD Operations for buses

### **Description of Task to be completed**
It should have the following endpoint:
- GET /api/v1/buses

- GET /api/v1/buses?page={page}&limit={limit}

- POST /api/v1/buses

- GET /api/v1/buses/id

- PATCH /api/v1/buses/id

- DELETE /api/v1/buses/id

### **How should be manually tested?**
- Clone the repo
- checkout the branch `ft-roles-and-permissions-zqZkd9M8`
- Install all packages
- Set up dotenv with `.env.example` as reference
- Run this command `npm run migrateDB` to make migration
- You should use the credentials below or you may create your user to access some endpoint that need to be logged in
{
    "email":"admin@phantom.com",
    "password":"admin"
}
- Copy token and paste it in documentation or in header if you are using postman.
- You may also test with the [documentation](https://phantom-caba-ft-crud-fo-pasycn.herokuapp.com/api-docs/)


### **Relevant pivots tracker**
[#P6PsRgMK](https://trello.com/c/P6PsRgMK/11-crud-operations-for-buses)
